In [1]:
import pandas as pd

In [2]:
#Import data

totalcol = pd.read_csv('totalcol.csv')

In [3]:
#convert remaining categorical values to 0/1 dummy values

totalcol_origin = pd.get_dummies(totalcol)

# sort results of correlation by values
totalcol_origin.corr().unstack().sort_values().drop_duplicates()

sex_Female             sex_Male                -1.000000
libusea_Yes, < 12 Mo.  lib_frq_Unknown         -0.996996
eminuse_Yes            eminuse_No              -0.995356
intmob_No              intmob_Yes              -0.993893
has_visited            libusea_Never           -0.993294
disa_Yes               disa_No                 -0.991450
par_No                 par_Yes                 -0.989193
reg_Probably           reg_Definitely          -0.965264
bbhome1_Unknown        home3nw_Yes             -0.923283
lib_frq_Unknown        libusea_ordinal         -0.885244
bbhome1_Higher Speed   bbhome1_Unknown         -0.825835
eminuse_Yes            home3nw_Unknown         -0.808820
home3nw_Yes            eminuse_No              -0.802192
libusea_ordinal        libusea_Never           -0.798684
libuseb_Never          libuseb_Yes, < 12 Mo.   -0.794652
smart1_Yes             smart1_No               -0.768693
home3nw_Yes            home3nw_Unknown         -0.749032
eminuse_Yes            bbhome1_

In [6]:
#Married people visiting the library

x, y, hue = "marital", "prop", "has_visited"

prop_df = (df[x]
           .groupby(df[hue])
           .value_counts(normalize=True)
           .rename(y)
           .reset_index())

#print(prop_df)

print(totalcol['marital'].value_counts())
print(totalcol['has_visited'].value_counts())
print(totalcol['has_visited'].groupby(totalcol['marital']).value_counts())
print(prop_df)

Married          923
Never Married    501
Widowed          199
Divorced         195
Cohabitating     109
Separated         65
Refused            9
Unknown            3
Name: marital, dtype: int64
1    1639
0     365
Name: has_visited, dtype: int64
marital        has_visited
Cohabitating   1               87
               0               22
Divorced       1              154
               0               41
Married        1              782
               0              141
Never Married  1              404
               0               97
Refused        1                8
               0                1
Separated      1               44
               0               21
Unknown        1                2
               0                1
Widowed        1              158
               0               41
Name: has_visited, dtype: int64
    has_visited        marital      prop
0             0        Married  0.386301
1             0  Never Married  0.265753
2             0       Divo

<h3>First Example</h3><br>

<h4>1 sided z test with a significance level of 0.01</h4>
Are married people more likely to go to the library than not?<br>
<br>
<u>Values:</u><br>
Population 1: Married and hasn't visited<br>
Proportion= .386301<br>
n = 365<br>
<br>
Population 2: Married and has visited<br>
Proportion -= .47712<br>
n = 1639<br>
<br>
$H_{0}$ = $P_{1}$ >= $P_{2}$<br>
$H_{alt}$ = $P_{1}$ < $P_{2}$<br>


$\ p = \frac{(p1 * n1 + p2 * n2)}{(n1 + n2)}$<br>

$\ p = \frac{(.386301 * 365 + .47712 * 1639)}{365+1639}$<br>

p=0.4606<br>
<br>
<h4> Standard Error</h4>
$\ SE = \sqrt{p(1-p) * (\frac{1}{n1} + \frac{1}{n2})}$<br>

$\ SE = \sqrt{(0.4606(1-.4606) * (\frac{1}{365} + \frac{1}{1639}))}$<br>
<br>
SE = 0.0288<br>
<br>
<h4> z score </h4>
$\ z = p1 - \frac{p2}{SE}$<br>
$\ z = \frac{.3863 - .47712}{.0288}$<br>
z = -3.15

Probability of a z value of -3.15 = 0.0008<br>
0.0008 is less than 0.01.  The null hypothesis is <b>rejected</b>.<br>
Married people are more likely to go have visited the library than not.